RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc

```diff
@@ -140,12 +139,11 @@ void SeedGeneratorFromProtoTracksEDProducer::produce(edm::Event& ev, const edm::
         GlobalTrackingRegion region(mom_perp, vtx, 0.2, 0.2);
 
         seedCreator_.init(region, es, nullptr);
-        seedCreator_.makeSeed(
-            *result,
-            SeedingHitSet(hits[0],
-                          hits[1],
-                          hits.size() > 2 ? hits[2] : SeedingHitSet::nullPtr(),
-                          (includeFourthHit_ && hits.size() > 3) ? hits[3] : SeedingHitSet::nullPtr()));
+
+        if (hits.size() == 4 and not includeFourthHit_)
```

Comment:
re-posting, hopefully in a way that github would not hide the comment

based on https://github.com/cms-sw/cmssw/pull/36184#issuecomment-1009989757
I guess that this line should change to 
```suggestion
        if (hits.size() > 3 and not includeFourthHit_)
```
to be consistent with the old implementation.
This is mainly to keep this PR technical.

Reply:
Sorry, yes it was hidden to me.  Consider that `includeFourthHit_` is set to `True` for the latest HLT setups (` 11834.0` included) so we would still end up running the other branch and would still see the differences. If we want this PR to stick to technical without adding any extra flags I'd go with something like


```
if (hits.size() > 3)
{
  if(includeFourthHit_)
    seedCreator_.makeSeed(*result, {hits[0], hits[1], hits[2],hits[3]});
  else
    seedCreator_.makeSeed(*result, {hits[0], hits[1], hits[2]});
} else 
{
   seedCreator_.makeSeed(*result, hits);
}
```

Reply:
> If we want this PR to stick to technical without adding any extra flags I'd go with something like ...

Thanks for clarifying that my proposed edit would not make this PR technical. 
IIRC, the `includeFourthHit_` was not meant to be exactly for the 4th hit anyway; it was supposed to cover any extension beyond the 3rd hit.

In the current implementation, if `includeFourthHit_` is false and a seed has 5 hits, all hits will be included, but a 4-hit seed will be truncated to 3 hits. This is awkward.

So, I think that my proposed edit still applies.

Reply:
Good for me, probably I'd change the name of `includeFourthHit_ ` in a following PR to solve the ambiguity.